cdap-cli/src/main/java/co/cask/cdap/cli/command/SetPreferencesCommand.java

```diff
@@ -47,19 +48,26 @@ public void printSuccessMessage(PrintStream printStream, ElementType type) {
 
   @Override
   public void perform(Arguments arguments, PrintStream printStream) throws Exception {
-    String runtimeArgs = arguments.get(ArgumentName.RUNTIME_ARGS.toString());
-    Map<String, String> args = ArgumentParser.parseMap(runtimeArgs, ArgumentName.RUNTIME_ARGS.toString());
-    setPreferences(arguments, printStream, args);
+    String[] programIdParts = new String[0];
+    if (arguments.hasArgument(type.getArgumentName().toString())) {
+      programIdParts = arguments.get(type.getArgumentName().toString()).split("\\.");
+      if (programIdParts.length < 2) {
```

Comment:
it does still work for instance and namespace since they won't have  a program ID so it'll skip this part. this is here because when "v1=k1 v2=k2" and myApp.myFlow is in the wrong order, it will give "Error: preferences must be of the form 'key1=val1 key2=val2'" instead of "Error: Invalid input. Expected format: set workflow preferences <app-id.workflow-id> <preferences>"